pallets/ah-migrator/src/scheduler.rs

```diff
@@ -0,0 +1,115 @@
+// Copyright (C) Parity Technologies (UK) Ltd.
+// This file is part of Polkadot.
+
+// Polkadot is free software: you can redistribute it and/or modify
+// it under the terms of the GNU General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+
+// Polkadot is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU General Public License for more details.
+
+// You should have received a copy of the GNU General Public License
+// along with Polkadot.  If not, see <http://www.gnu.org/licenses/>.
+
+use crate::*;
+use frame_support::traits::DefensiveTruncateFrom;
+use pallet_rc_migrator::scheduler::{alias::Scheduled, RcSchedulerMessage};
+
+/// Messages set from the RC Migrator concerning the Scheduler pallet.
+pub type RcSchedulerMessageOf<T> = RcSchedulerMessage<BlockNumberFor<T>, RcScheduledOf<T>>;
+
+/// Relay Chain `Scheduled` type.
+// From https://github.com/paritytech/polkadot-sdk/blob/f373af0d1c1e296c1b07486dd74710b40089250e/substrate/frame/scheduler/src/lib.rs#L203
+pub type RcScheduledOf<T> =
+	Scheduled<call::BoundedCallOf<T>, BlockNumberFor<T>, <T as Config>::RcPalletsOrigin>;
+
+impl<T: Config> Pallet<T> {
+	pub fn do_receive_scheduler_messages(
+		messages: Vec<RcSchedulerMessageOf<T>>,
+	) -> Result<(), Error<T>> {
+		log::info!(target: LOG_TARGET, "Processing {} scheduler messages", messages.len());
+		Self::deposit_event(Event::SchedulerMessagesReceived { count: messages.len() as u32 });
+		let (mut count_good, mut count_bad) = (0, 0);
+
+		for message in messages {
+			match Self::do_process_scheduler_message(message) {
+				Ok(()) => count_good += 1,
+				Err(_) => count_bad += 1,
+			}
+		}
+
+		Self::deposit_event(Event::SchedulerMessagesProcessed { count_good, count_bad });
+		log::info!(target: LOG_TARGET, "Processed {} scheduler messages", count_good);
+
+		Ok(())
+	}
+
+	fn do_process_scheduler_message(message: RcSchedulerMessageOf<T>) -> Result<(), Error<T>> {
+		log::debug!("Processing scheduler message: {:?}", message);
+
+		match message {
+			RcSchedulerMessage::IncompleteSince(block_number) => {
```

Comment:
The assumption is that we use Relay block numbers and an updated scheduler paller, right?

Reply:
On AH currently we only need a scheduler with RC block number provider. It will serve the referenda. For now I do not see any other use cases for the scheduler. 
Why RC block provider for referenda? Because for referendums we want the times as decision period to be close to real world period of 28 days for example.

Reply:
Discussed in DM.